crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -10,32 +12,106 @@ use super::{
     tx_digest::TransactionDigest,
 };
 use async_graphql::*;
+use sui_json_rpc_types::{
+    SuiExecutionStatus, SuiTransactionBlockDataAPI, SuiTransactionBlockEffects,
+    SuiTransactionBlockEffectsAPI,
+};
+use sui_sdk::types::digests::TransactionDigest as NativeTransactionDigest;
+
+#[derive(Clone)]
+pub(crate) struct TransactionBlock(pub sui_json_rpc_types::SuiTransactionBlockResponse);
+
+#[Object]
+impl TransactionBlock {
+    async fn digest(&self) -> TransactionDigest {
+        TransactionDigest::from_array(self.0.digest.into_inner())
+    }
+
+    async fn effects(&self) -> Result<Option<TransactionBlockEffects>> {
+        let tx_effects = self.0.effects.as_ref();
+
+        Ok(Some(TransactionBlockEffects {
+            digest: self.0.digest,
```

Comment:
Porting over the comment from above -- this shouldn't be the tx digest, it should be the effects digest.